packages/lime-proto-olsr/src/olsr.lua

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/lua
+
+local network = require("lime.network")
+local config = require("lime.config")
+local fs = require("nixio.fs")
+local libuci = require("uci")
+local wireless = require("lime.wireless")
+local utils = require("lime.utils")
+
+olsr = {}
+
+olsr.configured = false
+
+function olsr.configure(args)
+	if olsr.configured then return end
+	olsr.configured = true
+
+	local uci = libuci:cursor()
+	local ipv4 = network.primary_address()
+
+	fs.writefile("/etc/config/olsrd", "")
+
+	uci:set("olsrd", "lime", "olsrd")
+	uci:set("olsrd", "lime", "LinkQualityAlgorithm", "etx_ff")
+	uci:set("olsrd", "lime", "IpVersion", "4")
+
+    uci:set("olsrd", "limejson", "LoadPlugin")
+    uci:set("olsrd", "limejson", "library", "olsrd_jsoninfo.so.0.0")
+    uci:set("olsrd", "limejson", "accept", "127.0.0.1")
+
+	uci:set("olsrd", "limehna", "Hna4")
+	uci:set("olsrd", "limehna", "netaddr", ipv4:network():string())
+	uci:set("olsrd", "limehna", "netmask", ipv4:mask():string())
+
+	uci:save("olsrd")
+
+
+end
+
+function olsr.setup_interface(ifname, args)
+	if ifname:match("^wlan%d+_ap") then return end
+	vlanId = args[2] or 23
```

Comment:
default vlan-id of protocols should be < 16 see discussion at https://github.com/libre-mesh/lime-packages/pull/10